apps/desktop/src/app/upload/select-files/select-files.component.ts

```diff
@@ -128,4 +130,35 @@ export class SelectFilesComponent implements AfterViewInit, OnDestroy {
     this.selectionChange.emit(this.selection);
     this.cdr.detectChanges();
   }
+
+  toggleSelectAll(selectAll: boolean) {
+    this.isSelectingAll = selectAll;
+    if (this.isSelectingAll) {
+      this.loading = true;
+      this.cdr?.markForCheck();
+      this.getAllFiles().subscribe((items) => {
+        this.selection = new SelectionModel(true, items);
+        this.selectionChange.emit(this.selection);
+        this.loading = false;
+        this.cdr?.markForCheck();
+      });
+    } else {
+      this.selection = new SelectionModel(true, []);
+      this.selectionChange.emit(this.selection);
+      this.cdr?.markForCheck();
+    }
+  }
+
+  getAllFiles(): Observable<SyncItem[]> {
+    return (this.source as ISourceConnector).getFiles().pipe(
```

Comment:
It selects the first two pages of results. Should it get more ? or two is enough ?